File: TicTacToeApp.API/Entity/Game.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using TicTacToeApp.API.Entity.Enums;

namespace TicTacToeApp.API.Entity;

public class Game
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public StatusGame Status { get; set; } = StatusGame.Active;
    
    // [NotMapped]
    public string?[][] Board { get; set; } = null!;
    public ResultGame Result { get; set; } = ResultGame.None;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public Player CurrentMove { get; set; } = Player.X;
    public uint CurrentStep { get; set; } = 0;
    
    // public User User { get; set; } = default!;
    // public int UserId { get; set; }
}

File: TicTacToe.Angular/src/models/game.ts
export interface Game {
  id: string;
  status: StatusGame;
  board: string[][] | null;
  result: ResultGame;
  createdAt: string;
  currentMove: Player;
  currentStep: number;
}

export enum StatusGame {
  Active = "Active",
  Complete = "Complete"
}

export enum ResultGame {
  None = "None",
  XWon = "XWon",
  OWon = "OWon",
  Draw = "Draw"
}

export enum Player {
  X = 'X',
  O = 'O'
}
